spec/models/service_spec.rb

```diff
@@ -53,4 +53,52 @@
       expect(service.required_tiers("all")).to eq([2, 3, 1])
     end
   end
+
+  describe "#duplicate!" do
+    it "duplicates the service" do
+      service = create(:service, :all_tiers, lgsl_code: 1)
+
+      interaction1 = create(:interaction, lgil_code: 1, label: "interaction1")
+      service_interaction1 = create(:service_interaction, service: service, interaction: interaction1)
+      create(:link, service_interaction: service_interaction1, url: "https://example.com")
+      create(:link, service_interaction: service_interaction1, url: "https://example.com/test")
+      create(:link, service_interaction: service_interaction1, url: "https://example.com/test/again")
+
+      interaction2 = create(:interaction, lgil_code: 2, label: "interaction2")
+      service_interaction2 = create(:service_interaction, service: service, interaction: interaction2)
+      create(:link, service_interaction: service_interaction2, url: "https://example.com")
+      create(:link, service_interaction: service_interaction2, url: "https://example.com/test")
+      create(:link, service_interaction: service_interaction2, url: "https://example.com/test/again")
+
+      service.duplicate!(2)
+
+      new_service = Service.find_by(lgsl_code: 2)
+
+      expect(new_service.label).to eq("TRANSITION -- #{service.label}")
+      expect(new_service.slug).to eq("TRANSITION -- #{service.slug}")
+
+      expect(service.interactions.count).to eq(2)
+      expect(service.service_interactions.count).to eq(2)
+      expect(service.links.count).to eq(6)
```

Comment:
These assertions seem a bit strange as they're essentially testing the factory works and not the method under test. It'd seem to make more sense to test: `expect(new_service.interactions.count).to eq(service.interactions.count)`

Reply:
Thanks - fixed up as per your suggestion